src/library/baseplaylistfeature.cpp

```diff
@@ -650,22 +650,23 @@ QModelIndex BasePlaylistFeature::constructChildModel(int selected_id) {
     return m_childModel.index(selected_row, 0);
 }
 
-void BasePlaylistFeature::updateChildModel(int selected_id) {
-    buildPlaylistList();
+void BasePlaylistFeature::updateChildModel(int playlistId) {
+    // the following call fetches whole m_playlistList.
+    // Anything may have change due to a concurrent Mixxx instance.
+    updatePlaylistList(playlistId);
 
     int row = 0;
     for (QList<QPair<int, QString> >::const_iterator it = m_playlistList.begin();
          it != m_playlistList.end(); ++it, ++row) {
         int playlist_id = it->first;
         QString playlist_name = it->second;
 
-        if (selected_id == playlist_id) {
+        if (playlistId == playlist_id) {
```

Comment:
The same name in camel-case and snake case notation. Who should understand this code?